packages/beacon-node/src/chain/chain.ts

```diff
@@ -637,6 +641,37 @@ export class BeaconChain implements IBeaconChain {
 
   private onForkChoiceFinalized(this: BeaconChain, cp: CheckpointWithHex): void {
     this.logger.verbose("Fork choice finalized", {epoch: cp.epoch, root: cp.rootHex});
+    if (this.isWSCheckpointValidationPending && this.opts.forwardWSCheckpoint) {
+      try {
+        this.isWSCheckpointValidationPending = !this.forkChoice.verifyForwardCheckpoint(this.opts.forwardWSCheckpoint);
+        if (!this.isWSCheckpointValidationPending) {
+          const finalizedState = this.checkpointStateCache.get(cp);
+          if (finalizedState === null) {
+            throw new Error("finalizedState not found!");
+          }
+          const wssCheck = isWithinWeakSubjectivityPeriod(
+            this.config,
+            finalizedState,
+            this.opts.forwardWSCheckpoint,
+            false
+          );
+          const wsData = {
+            epoch: this.opts.forwardWSCheckpoint.epoch,
+            root: toHex(this.opts.forwardWSCheckpoint.root),
+            isWithinWeakSubjectivityPeriod: wssCheck,
+          };
+          if (wssCheck) {
+            this.logger.info("forwardWSCheckpoint found and valid", wsData);
+          } else {
+            this.logger.warn("forwardWSCheckpoint found on chain but is stale, recommend using a fresh one", wsData);
+          }
+        }
+      } catch (error) {
+        // Critical error, exit the chain
+        this.processShutdownCallback(error as Error);
```

Comment:
Being such a critical action (kill lodestar completely) putting here in a generic try / catch feels too dangerous. I would really like to do
```ts
if (condition) {
  this.processShutdownCallback(error as Error);
}
```